### DISCUSSION TITLE
make same props reusable

### DISCUSSION THREAD

**Discussion by hi-reeve** [cdmoro/bootstrap-vue-3#167] [Ideas]
this is not issue, i just kinda want suggesting the way props is created. i notice that there is so many component have the same props. instead of declaring it one by one per component. why don't we create a composable to create the props itself?
i got this idea from how vuetify team created their props

here is the reference : 
https://github.com/vuetifyjs/vuetify/blob/next/packages/vuetify/src/composables/border.ts

example of how it is implemented
https://github.com/vuetifyjs/vuetify/blob/next/packages/vuetify/src/components/VBtn/VBtn.tsx

i'm willing to help if this proposal will move further.

**Comment by cdmoro** [comment]
It seems really interesting! I'll convert this issue into a discussion, so we can debate this feature.

**Comment by hi-reeve** [comment]
thankyou! i will try as much as i can help.
